File: react_native_app/src/core/usecase/usecase.ts
// Contrato base de un caso de uso (Clean Architecture).
// `T` es el tipo de retorno; `P` el tipo de entrada (parámetros).
export interface UseCase<T, P> {
  call(params: P): Promise<T>;
}

/** Marcador para casos de uso que no reciben parámetros. */
export type NoParams = void;

File: flutter_app/lib/core/usecase/usecase.dart
/// Contrato base de un caso de uso (Clean Architecture).
/// [T] es el tipo de retorno; [P] el tipo de entrada (parámetros).
abstract interface class UseCase<T, P> {
  Future<T> call(P params);
}

/// Marcador para casos de uso que no reciben parámetros.
class NoParams {
  const NoParams();
}
